yalafi/packages/cleveref.py

```diff
@@ -0,0 +1,215 @@
+
+#
+#   YaLafi module for LaTeX package cleveref
+#
+
+from yalafi.defs import InitModule, Macro, VoidToken
+from yalafi import utils
+import re
+
+
+require_packages = []
+
+# Regular Expressions used to read sed file:
+re_ref = re.compile(r'''
+s/\\
+(                   # group 1: '\cref' or '\Cref'
+    \\cref|\\Cref
+)
+(?:\\)?             # non capturing group
+(\*?)               # group 2: '*' if present, '' else
+\{
+    ([^\}\{]+)      # group 3: the label of the reference
+\}
+/
+    (.*)            # group 4: the replacement string
+/g
+''', re.VERBOSE)
+re_ref_range = re.compile(r'''
+s/\\
+(                   # group 1: '\cref' or '\Cref'
+    \\crefrange|\\Crefrange
+)
+(?:\\)?             # non capturing group
+(\*?)               # group 2: '*' if present, '' else
+\{
+    ([^\}\{]+)      # group 3: the label of the reference
+\}
+/
+    (.*)            # group 4: the replacement string
+/g
+''', re.VERBOSE)
+re_command = re.compile(r'''
+s/\\
+(                   # group 1: command
+    \\
+    (               # group 2: capturing cC
+        \[cC\]
+    )?
+    [a-zA-Z@]+
+)
+(                   # group 3: possible arguments
+    (?:\{\.\*\})+
+)?
+\s*                 # possible whitespaces inbetween
+/
+    (.*)            # group 4: replacement
+/g
+''', re.VERBOSE)
+re_cC = re.compile(r'(\[cC\])')
+re_escaped_dot = re.compile(r'\\\.')
+re_escaped_backslash = re.compile(r'\\\\')
+re_escaped_star = re.compile(r'\\\*')
+
+
+def re_remove_escaped_symbols(string):
+    r"""
+    Replaces certain strings that are escaped in sed file:
+    '\\.' -> '.'
+    '\\\\' -> '\\'
+    '\*' -> ''
+    """
+    string = re_escaped_dot.sub('.', string)
+    string = re_escaped_star.sub('', string)
+    string = re_escaped_backslash.sub(r'\\', string)
+    return string
+
+
+macro_read_sed = '\\YYCleverefInput'

```

Comment:
IMHO, this name definition should be shifted to the top, e.g., below `require_packages`. I'm undecided about  the following error message templates.